### PR TITLE
Add monitoring config and variable to rook component

### PR DESCRIPTION
**Pull request by surajssd**

### CODE REVIEW

File: pkg/components/rook/component.go [hunk 1]
```diff
@@ -40,6 +40,7 @@ type component struct {
 	AgentTolerationEffect    string `hcl:"agent_toleration_effect,optional"`
 	DiscoverTolerationKey    string `hcl:"discover_toleration_key,optional"`
 	DiscoverTolerationEffect string `hcl:"discover_toleration_effect,optional"`
+	EnableMonitoring         bool   `hcl:"enable_monitoring,optional"`
```
Comment (invidian) [inline pkg/components/rook/component.go:43]: I'm on the fence with `enable_monitoring` here. We use `service_monitor` for some components, `metrics` for others. `enable_monitoring` make sense here, but inconsistency is bothering.

Comment (invidian) [reply pkg/components/rook/component.go:43]: Created an issue for it https://github.com/kinvolk/lokomotive/issues/407.

Comment (surajssd) [reply pkg/components/rook/component.go:43]: Eventually we should change all of them to `enable_monitoring`. We have to start somewhere.

**Comment by surajssd** [comment]
@invidian 

> should we also add some tests for collecting metrics etc?

It is not that simple to add e2e monitoring tests for this. So right now we install only `rook` component and not `rook-ceph` component. The metrics are emitted by _manager_ pod and the _OSD_ pods. And these pods only start once we have `rook-ceph` installed. Hence testing metrics does not make sense, since there are no metrics. The tests will simply fail. We can't test `testComponentsPrometheusMetrics` right now. But we can test `testComponentAlerts`(I have added test for this in last commit). And test `testScrapeTargetRechability` is automatic so we don't need to worry about it right now.

File: pkg/components/rook/component.go (hunk 1 above)
Comment (invidian) [reply pkg/components/rook/component.go:43]: Still not sure as mentioned in the issue, but it seems reasonable, so it's fine for now IMO.

**Pull request merged**